sql/chunk.sql

```diff
@@ -35,3 +35,12 @@ CREATE OR REPLACE FUNCTION _timescaledb_internal.calculate_chunk_interval(
         dimension_coord BIGINT,
         chunk_target_size BIGINT
 ) RETURNS BIGINT AS '@MODULE_PATHNAME@', 'ts_calculate_chunk_interval' LANGUAGE C;
+
+-- Function for explicit chunk exclusion. Supply relation name/alias and an array of chunk ids as input.
+-- Intended to be used in WHERE clause.
+--
+-- Use it with care as this function directly affects what chunks are being scanned for data.
+-- Although this function is immutable (always returns true), we declare it here as volatile
+-- so PostgreSQL optimizer would not try to evaluate/reduce it in the planner phase
+CREATE OR REPLACE FUNCTION chunks_in(relation ANYELEMENT, chunks INTEGER[]) RETURNS BOOL
```

Comment:
Why is relation ANYELEMENT shouldnt this be regclass?

Reply:
This enables us to use aliases as well - with regclass it wouldn't be possible.

Reply:
Atm this seems to not be checked anywhere, so what is it used for?

Reply:
Not sure I understand what you mean, but this is being validated in `collect_quals_walker`.

Reply:
Then the type for relation should be RECORD, cause the way you are using it means you are passing the complete row into the function.

Reply:
No, that would not work because then you can't pass relation name as an argument.

Reply:
All the valid queries in your tests use RECORD. Could you add tests for passing in relation name.

Reply:
That's intentionally not allowed to prevent wrong input. The idea is to only allow passing relation/alias from the FROM clause.

Reply:
That contradicts your previous answer where you said it cant be RECORD because you want to pass in relation name.

Reply:
It's a relation name reference used in the FROM clause. Do you have a better naming proposal for the code comment?

Reply:
I've changed from relation name to relation